tests/device_buffer_tests.cu

```diff
@@ -73,7 +74,7 @@ TYPED_TEST(DeviceBufferTest, DefaultMemoryResource)
   EXPECT_EQ(this->size, buff.size());
   EXPECT_EQ(this->size, buff.ssize());
   EXPECT_EQ(this->size, buff.capacity());
-  EXPECT_EQ(rmm::mr::get_current_device_resource(), buff.memory_resource());
+  EXPECT_EQ(async_resource_ref{rmm::mr::get_current_device_resource()}, buff.memory_resource());
```

Comment:
question: Should this equality comparison work even without constructing an `async_resource_ref` from the `device_memory_resource*` returned by `get_current_device_resource()`? The implicit conversion from `device_memory_resource*` to `async_resource_ref` should kick in and hit the `operator==`, right?